extra-tests/classes/RollupFullRecalcTests.cls

```diff
@@ -1916,9 +1915,10 @@ private class RollupFullRecalcTests {
   static void shouldCacheRetrievedCalcItemsBetweenFullRecalcRuns() {
     // since we can't have more than one batch chunk run in the tests, but we need to test logic in the execute method
     // we'll call that method twice directly
+    RollupAsyncProcessor.additionalCalcItemCount = 1;
```

Comment:
What's the reason for setting `additionalCalcItemCount` here?

Reply:
previously, tests were not querying for additional records in the `retrieveAdditionalCalcItems` method in `RollupAsyncProcessor`. Now, not only are record counts being done by query, but also the actual queries themselves (when the count is greater than 0). This is a "shortcut" to only doing the fetching of additional items part, without explicitly doing the `Database.countQuery` part. That's why the number of queries goes up, below, as well, since the query to retrieve additional items is actually being done (instead of short-circuiting just within tests, which had its own problems that I've now addressed)

Reply:
That makes sense - it might be worth adding a comment in this test to explain that, but not strictly necessary